packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+import { isFunction, isUndefined } from '@aws-amplify/ui';
+import { usePaginate } from '../hooks/usePaginate';
+import { useStore } from '../../providers/store';
+import { useAction } from '../../do-not-import-from-here/actions';
+import { LocationData, LocationItemData } from '../../actions';
+
+interface UseLocationDetailView {
+  hasNextPage: boolean;
+  hasError: boolean;
+  isLoading: boolean;
+  isPaginateNextDisabled: boolean;
+  isPaginatePreviousDisabled: boolean;
+  message: string | undefined;
+  pageItems: LocationItemData[];
+  page: number;
+  onAccessItem: (location: LocationData) => void;
+  onRefresh: () => void;
+  onPaginateNext: () => void;
+  onPaginatePrevious: () => void;
+  onAddFiles: (files: File[]) => void;
+}
+
+export type LocationDetailViewActionType =
+  | { type: 'REFRESH_DATA' } // refresh data only
+  | { type: 'RESET' } // reset view to initial state
+  | { type: 'PAGINATE_NEXT' }
+  | { type: 'PAGINATE_PREVIOUS' }
+  | { type: 'PAGINATE'; page: number }
+  | { type: 'ACCESS_ITEM'; key: string }
+  | { type: 'NAVIGATE'; location: LocationData }
+  | { type: 'ADD_FILES'; files: File[] }
+  | { type: 'SEARCH'; query: string; includeSubfolders?: boolean };
+
+interface InitialValues {
+  pageSize?: number;
+  delimiter?: string;
+}
+
+export interface UseLocationDetailViewOptions {
+  initialValues?: InitialValues;
+  onDispatch?: React.Dispatch<LocationDetailViewActionType>;
+  onActionSelect?: (type: string) => void;
+  onExit?: () => void;
```

Comment:
I wonder if this should be something more akin to `onNavigateAway` - "exiting" a component is a little strange to me

Reply:
it think the intention is to 'exit' a view, but I think we could consider changing that in a separate PR as `onExit` is used in a few different places.

Reply:
Don't disagree that it's a little weird, but I think we should keep onExit for now, as that's what we landed on as best option in the fit and finish review.

Reply:
We need to align all of the disparate names in these hooks. Can be done in a handled in a follow up since it shouldn't impact logic